src/quote/swap/adapters/lifi/index.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  ContractCallsQuoteRequest,
+  getContractCallsQuote,
+  getQuote,
+  LiFiStep,
+} from '@lifi/sdk'
+
+import {
+  SwapQuote,
+  SwapQuoteProvider,
+  SwapQuoteRequest,
+} from 'quote/swap/interfaces'
+
+import { getSwapData } from './swap-data'
+
+export class LiFiSwapQuoteProvider implements SwapQuoteProvider {
+  constructor(readonly apiKey: string, readonly integrator: string) {}
+
+  async getSwapQuote(request: SwapQuoteRequest): Promise<SwapQuote | null> {
+    // This is not ideal but right now the only way to get only uniV2 and sushi quotes
+    const allowExchanges = ['uniswap', 'sushiswap']
```

Comment:
This would likely be an issue in production as LiFi only supports UniV2

Reply:
e.g. this won't work since `sushiswap` is provided?

Reply:
Since the contracts require UniV3 quotes